packages/features/form-builder/FormBuilder.tsx

```diff
@@ -0,0 +1,734 @@
+                  update(fieldDialog.fieldIndex, data);
+                } else {
+                  const field: RhfFormField = {
+                    ...data,
+                    sources: [
+                      {
+                        label: "User",
+                        type: "user",
+                        id: "user",
+                        fieldRequired: data.required,
+                      },
+                    ],
+                  };
+                  field.editable = field.editable || "user";
+                  append(field);
+                }
+                setFieldDialog({
+                  isOpen: false,
+                  fieldIndex: -1,
+                });
+              }}>
+              <SelectField
+                required
+                isDisabled={
+                  fieldForm.getValues("editable") === "system" ||
+                  fieldForm.getValues("editable") === "system-but-optional"
+                }
+                onChange={(e) => {
+                  const value = e?.value;
+                  if (!value) {
+                    return;
+                  }
+                  fieldForm.setValue("type", value);
+                }}
+                value={FieldTypes.find((fieldType) => fieldType.value === fieldForm.getValues("type"))}
+                options={FieldTypes.filter((f) => !f.systemOnly)}
+                label="Input Type"
+              />
+              <InputField
+                required
+                {...fieldForm.register("name")}
+                containerClassName="mt-6"
+                disabled={
+                  fieldForm.getValues("editable") === "system" ||
+                  fieldForm.getValues("editable") === "system-but-optional"
+                }
+                label="Name"
+              />
+              <InputField
+                {...fieldForm.register("label")}
+                // System fields have a defaultLabel, so there a label is not required
+                required={!["system", "system-but-optional"].includes(fieldForm.getValues("editable") || "")}
+                placeholder={t(fieldForm.getValues("defaultLabel") || "")}
+                containerClassName="mt-6"
+                label="Label"
+              />
+              {fieldType?.isTextType ? (
+                <InputField
+                  {...fieldForm.register("placeholder")}
+                  containerClassName="mt-6"
+                  label="Placeholder"
+                  placeholder={t(fieldForm.getValues("defaultPlaceholder") || "")}
+                />
+              ) : null}
+
+              {fieldType?.needsOptions ? (
+                <Controller
+                  name="options"
+                  render={({ field: { value, onChange } }) => {
+                    return <OptionsField onChange={onChange} value={value} className="mt-6" />;
+                  }}
+                />
+              ) : null}
+              <Controller
```

Comment:
When editing a field I can't click 'yes' or 'no', it's working when creating a new field

Reply:
@CarinaWolli I think you must be editing a system field, that doesn't allow marking itself optional. User fields can be marked yes or not in edit mode.

Reply:
I see, it works with the fields that I created. We should make that more clear that this is disabled and can't be edited

Reply:
Yeah, I am not sure how to do it. But it is a valid point. @Jaibles I think a better disabled state for yes No button should help I guess.

Reply:
Agreed, I too saw the problem. I was thinking that all buttons should be there but they should be disabled.

Reply:
Yes I think I like that better 👍 I just starts looking pretty messy once you change the order and you have system fields and your fields mixed up